src/Illuminate/Routing/Router.php

```diff
@@ -360,6 +361,21 @@ public function resource($name, $controller, array $options = [])
     }
 
     /**
+     * Register a new redirect route with the router.
+     *
+     * @param  string  $from
+     * @param  string  $to
+     * @param  int  $status
+     * @return \Illuminate\Routing\Route
+     */
+    public function redirect($from, $to, $status = 301)
+    {
+        return $this->get($from, function () use ($to, $status) {
+            return new RedirectResponse($to, $status);
+        });
```

Comment:
It's a bit of a stretch, but I have a solution.

Controllers can specify an optional `callAction` method which will be called _if present_. The method is given the `$method` and `$parameters` as it's arguments, additionally, we could provide the `Illuminate\Routing\Route` instance when calling it in `Illuminate\Routing\ControllerDispatcher::dispatch`. By doing so, we would have access to the routes `getAction()` method, which allows us to access `data` from the following:

```php
public function redirect($uri, $destination, $status = 301)
{
    return $this->any($uri, [
        'uses' => '\Illuminate\Foundation\Http\RedirectController@handle`,
        'data'' => compact('destination', 'status'),
    ]);
}
```

The `callAction` method would look like such:

```php
public function callAction($method, $parameters, $route)
{
    return $this->handle(
        $route->getAction()['data']['destination'],
        $route->getAction()['data']['status']
    );
}
```